builtin/providers/aws/resource_aws_route53_record.go

```diff
@@ -264,14 +347,23 @@ func resourceAwsRoute53RecordRead(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	d.Set("ttl", record.TTL)
-	// Only set the weight if it's non-nil, otherwise we end up with a 0 weight
-	// which has actual contextual meaning with Route 53 records
-	//   See http://docs.aws.amazon.com/fr_fr/Route53/latest/APIReference/API_ChangeResourceRecordSets_Examples.html
+
+	if record.Failover != nil {
+		d.Set("failover_routing_policy.type", record.Failover)
+	}
+	if record.GeoLocation != nil {
+		d.Set("geolocation_routing_policy.continent", record.GeoLocation.ContinentCode)
+		d.Set("geolocation_routing_policy.country", record.GeoLocation.CountryCode)
+		d.Set("geolocation_routing_policy.subdivision", record.GeoLocation.SubdivisionCode)
+	}
+	if record.Region != nil {
+		d.Set("latency_routing_policy.region", record.Region)
```

Comment:
There are issues here, in that we are not checking the error returned from `d.Set` here. For simple values of `TypeInt` or `TypeString` this is OK, however for complicated things like `TypeSet` or `TypeList` we need to check for errors. 

Here I know we're erring, as you can't do a deep set like this (`geolocation_routing_policy.subdivision`) you must set the entire `geolocation_routing_policy` set. As a result we're seeing a perpetual plan here.